### PR TITLE
Replicas: Add API and python client api for declaring replicas bad by RSE and DIDs #4099

### PR DESCRIPTION
**Pull request by ericvaandering**
Pull request title
------------------

The format of the Pull request title must be:

    <component>: <short_change_message> #<issue number>

If you add a [github-recognised keyword](https://help.github.com/articles/closing-issues-using-keywords/) then
the associated issue can be closed automatically once the pull request is merged, e.g.::

    <component>: <short_change_message> Fix #<issue number>

Valid component names are listed in the `label list <https://github.com/rucio/rucio/labels>`_.

### CODE REVIEW

File: lib/rucio/api/replica.py [hunk 1]
```diff
@@ -400,6 +400,31 @@ def add_bad_pfns(pfns, issuer, state, reason=None, expires_at=None, vo='def'):
     return replica.add_bad_pfns(pfns=pfns, account=issuer, state=state, reason=reason, expires_at=expires_at)
 
 
+def add_bad_dids(dids, rse, issuer, state, reason=None, expires_at=None, vo='def'):
+    """
+    Add bad PFNs.
+
+    :param pfns: the list of new files.
```
Comment (cserf) [inline lib/rucio/api/replica.py]: :param dids:

**Review by cserf — COMMENTED** [review]
Looks good to me. Just a small typo in the docstring

File: lib/rucio/api/replica.py (hunk 1 above)
Comment (ericvaandering) [reply lib/rucio/api/replica.py]: Fixed

**Comment by ericvaandering** [comment]
Please re-review and approve.

**Review by bari12 — APPROVED** [review]

**Pull request merged**